packages/server/test/inference.test.ts

```diff
@@ -16,6 +17,25 @@ test('infer input & output', async () => {
       async resolve({ input }) {
         return { input };
       },
+    })
+    .query('withOutput', {
+      input: z.string(),
+      output: z.object({
+        input: z.string(),
+      }),
```

Comment:
Maybe tests to do with this should be in it's own file? Like `output-parser.test.ts`

Reply:
I have kept these high-level type inference related tests in this file. However, I have also added `outputParser.test.ts` with a more comprehensive set of tests relating to this PR, hope this is ok. 👍